src/parser/bison_parser.y

```diff
@@ -689,33 +691,40 @@ select_statement:
 			$$ = $2;
 			$$->withDescriptions = $1;
 		}
-	|	opt_with_clause select_with_paren set_operator select_paren_or_clause opt_order opt_limit {
-			// TODO: allow multiple unions (through linked list)
-			// TODO: capture type of set_operator
-			// TODO: might overwrite order and limit of first select here
+	|	opt_with_clause select_with_paren set_operator select_statement_in_set_operation opt_order opt_limit {
 			$$ = $2;
+			if ($$->setOperators == nullptr) {
+				$$->setOperators = new std::vector<SetOperator*>();
```

Comment:
Is this really consistent with what we do in other places? I feel that pointers are overused in the parser anyway, but if we set it to an empty vector here and never have to check if this is a nullptr (i.e., unset), what is the point of the indirection?

Reply:
It's consistent in the sense that all the other vectors are also initialized in the parser_definition and not during initialization (afaik). The nullptr check is not necessary for any other rule

Reply:
Are we talking about the same thing? I see a couple of places where a vector is explicitly nullptr:

```
opt_column_list:
		'(' ident_commalist ')' { $$ = $2; }
	|	/* empty */ { $$ = nullptr; }
	;
```

```
opt_order:
		ORDER BY order_list { $$ = $3; }
	|	/* empty */ { $$ = nullptr; }
	;

order_list:
		order_desc { $$ = new std::vector<OrderDescription*>(); $$->push_back($1); }
	|	order_list ',' order_desc { $1->push_back($3); $$ = $1; }
	;
```

It looks like `vector<[^(]*>\(\)` is usually followed by a push_back.

Reply:
Well the order list is initially set as nullptr as well. They just don't need the check because they have a very specific base case.

I'm fully in favor of doing the initialization in the constructor, however it would be inconsistent with the rest of the code-base. If you give the okay for the inconsistency, I have no problem to change it

Reply:
Are you in the office right now? Maybe this is easier to discuss offline.

Reply:
Saw this too late. You seem to be gone now...

Are you working this weekend again?

Reply:
Yes, but not in the office